.drone.yml

```diff
@@ -22,13 +22,29 @@ steps:
       - "hostname=$(curl -s -H \"X-aws-ec2-metadata-token: $IMDS_TOKEN\" http://169.254.169.254/latest/meta-data/public-hostname)"
       - echo "Running on $hostname"
 
-  - name: clone
+  # We cache a clone of staging in the `cache-staging-clone` pipeline
+  - name: get-cached-staging
+    image: plugins/s3-cache
+    settings:
+      filename: cache-staging-clone.tar
+      endpoint: s3://code-dot-org-drone
+      restore: true
+      debug: true
+      pull: true
+      mount:
+        - .
+
+  - name: checkout
     image: bitnami/git
     commands:
-      - git lfs install
-      - git clone --branch $DRONE_SOURCE_BRANCH  --depth 200 $DRONE_GIT_HTTP_URL .
+      - git lfs install --skip-repo
+      # We should be in a cached git repo, but clone if the cache failed
+      - "[ -d .git ] || (echo 'WARNING: get-cached-staging failed: checkout will be SLOW and LFS $$$!' && git clone --branch $DRONE_BRANCH $DRONE_GIT_HTTP_URL .)"
```

Comment:
Fallback method so we're resilient (but eventually aware, thinking racking up LFS GB fast if caching fails) in case the cache file disappears from S3, the caching pipeline breaks, or something else fails

Reply:
Will this error surface anywhere useful?

Reply:
This shows up in the drone logs, its not an instant catch, but I think I phrased it so there's a chance when somebody notices the checkout step is 5 minutes slower while waiting for drone, they can tie this back to why its slower.